File: ProyectosJAVA/LP2SoftLogisticaController/src/pe/edu/pucp/lp2soft/almacen/mysql/ProductoMySQL.java
package pe.edu.pucp.lp2soft.almacen.mysql;
import java.sql.CallableStatement;
import java.sql.Connection;
import java.sql.ResultSet;
import java.util.ArrayList;
import pe.edu.pucp.lp2soft.almacen.dao.ProductoDAO;
import pe.edu.pucp.lp2soft.almacen.model.Producto;
import pe.edu.pucp.lp2soft.config.DBManager;
public class ProductoMySQL implements ProductoDAO{
    
    private Connection con;
    private ResultSet rs;
    private CallableStatement cs;
    
    @Override
    public ArrayList<Producto> listarPorNombre(String nombre) {
        ArrayList<Producto> productos = new ArrayList<>();
        try{
            con = DBManager.getInstance().getConnection();
            cs = con.prepareCall("{call LISTAR_PRODUCTOS_X_NOMBRE(?)}");
            cs.setString("_nombre",nombre);
            rs = cs.executeQuery();
            while(rs.next()){
                Producto producto = new Producto();
                producto.setIdProducto(rs.getInt("id_producto"));
                producto.setNombre(rs.getString("nombre"));
                producto.setUnidadMedida(rs.getString("unidad_medida"));
                producto.setPrecio(rs.getDouble("precio"));
                producto.setActivo(true);
                productos.add(producto);
            }
        }catch(Exception ex){
            System.out.println(ex.getMessage());
        }finally{
            try{con.close();}catch(Exception ex){System.out.println(ex.getMessage());}
        }
        return productos;
    }
}

File: ProyectosNET/LP2Soft/LP2SoftLogisticaController/Almacen/MySQL/ProductoMySQL.cs
using LP2SoftDBManager;
using LP2SoftLogisticaController.Almacen.DAO;
using LP2SoftLogisticaModel.Almacen;
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LP2SoftLogisticaController.Almacen.MySQL
{
    public class ProductoMySQL : ProductoDAO
    {
        private MySqlConnection con;
        private MySqlCommand comando;
        private MySqlDataReader lector;
        public BindingList<Producto> listarPorNombre(string nombre)
        {
            BindingList<Producto> productos = new BindingList<Producto>();
            try
            {
                con = DBManager.Instance.getConnection();
                con.Open();
                comando = new MySqlCommand();
                comando.Connection = con;
                comando.CommandText = "LISTAR_PRODUCTOS_X_NOMBRE";
                comando.Parameters.AddWithValue("_nombre", nombre);
                comando.CommandType = System.Data.CommandType.StoredProcedure;
                lector = comando.ExecuteReader();
                while (lector.Read())
                {
                    Producto producto = new Producto();
                    if (!lector.IsDBNull(lector.GetOrdinal("id_producto"))) producto.IdProducto = lector.GetInt32("id_producto");
                    if (!lector.IsDBNull(lector.GetOrdinal("nombre"))) producto.Nombre = lector.GetString("nombre");
                    if (!lector.IsDBNull(lector.GetOrdinal("unidad_medida"))) producto.UnidadMedida = lector.GetString("unidad_medida");
                    if (!lector.IsDBNull(lector.GetOrdinal("precio"))) producto.Precio = lector.GetDouble("precio");
                    productos.Add(producto);
                }
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message);
            }
            finally
            {
                try { con.Close(); } catch (Exception ex) { throw new Exception(ex.Message); }
            }
            return productos;
        }
    }
}
